eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/iterator/DropWhileIterator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.lazy.iterator;
+
+import java.util.Iterator;
+
+import net.jcip.annotations.Immutable;
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.eclipse.collections.impl.block.predicate.DropWhileIterablePredicate;
+
+/**
+ * Iterates over the elements of the iterator skipping the first elements until predicate returns false.
+ */
+@Immutable
+public final class DropWhileIterator<T> implements Iterator<T>
+{
+    private final Iterator<T> delegateIterator;
+
+    public DropWhileIterator(Iterable<T> iterable, Predicate<? super T> predicate)
+    {
+        this(iterable.iterator(), predicate);
+    }
+
+    public DropWhileIterator(Iterator<T> iterator, Predicate<? super T> predicate)
+    {
+        this.delegateIterator = new SelectIterator<T>(iterator, new DropWhileIterablePredicate<T>(predicate));
```

Comment:
Yes. `DropWhileIterator` follows the exact same pattern as `DropIterator`. The deprecation would just re-direct to use:

`new SelectIterator<T>(iterable.iterator(), new DropIterablePredicate<T>(count))`